File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        int [] ans=new int[2];
        HashMap<Integer,Integer> mp=new HashMap<>();
        mp.put(nums[0],0);
        
        for(int i=1;i<nums.length;i++)
        {
            if(mp.containsKey(target-nums[i]))
            {
                ans[0]=mp.get(target-nums[i]);
                ans[1]=i;
                return ans;
            }
            mp.put(nums[i],i);
        }
        return ans;
        
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int,int> mp;
        
        vector<int> ans(2,0);
        
        for(int i=0;i<nums.size();i++)
        {
            if(mp.find(target-nums[i])!=mp.end())
            {
                ans[0]=mp[target-nums[i]];
                ans[1]=i;
                return ans;
            }
            
            mp[nums[i]]=i;
        }
        
        return ans;
        
    }
};